--- a/binutils/readelf.c
+++ b/binutils/readelf.c
@@ -375,7 +375,7 @@ static int
 fseek64 (FILE *stream, int64_t offset, int whence)
 {
-#if defined (HAVE_FSEEKO64)
-  off64_t o = offset;
+#if defined (HAVE_FSEEKO)
+  off_t o = offset;
   if (o != offset)
     {
       errno = EINVAL;
       return -1;
     }
-  return fseeko64 (stream, o, whence);
-#elif defined (HAVE_FSEEKO)
-  off_t o = offset;
+  return fseeko (stream, o, whence);
+#elif defined (HAVE_FSEEKO64)
+  off64_t o = offset;
   if (o != offset)
     {
       errno = EINVAL;
       return -1;
     }
-  return fseeko (stream, o, whence);
+  return fseeko64 (stream, o, whence);
 #else
   long o = offset;
   if (o != offset)
